src/shared/components/CSVExportButton.tsx

```diff
@@ -1,72 +1,102 @@
 // Libraries
 import React, {PureComponent} from 'react'
-import {connect} from 'react-redux'
+import {connect, ConnectedProps} from 'react-redux'
 
 // Components
-import {Button, ComponentStatus, IconFont} from '@influxdata/clockface'
-
-// Utils
-import {downloadTextFile} from 'src/shared/utils/download'
-import {getActiveTimeMachine} from 'src/timeMachine/selectors'
-import {createDateTimeFormatter} from 'src/utils/datetime/formatters'
+import {
+  Button,
+  ComponentStatus,
+  ComponentColor,
+  IconFont,
+} from '@influxdata/clockface'
+
+// Selectors and Actions
+import {getActiveQuery} from 'src/timeMachine/selectors'
+import {
+  runDownloadQuery,
+  DOWNLOAD_EVENT_COMPLETE,
+} from 'src/timeMachine/actions/queries'
 
 // Types
-import {AppState} from 'src/types'
+import {AppState, RemoteDataState} from 'src/types'
+
+type Props = ConnectedProps<typeof connector>
 
-interface StateProps {
-  files: string[] | null
+interface State {
+  state: RemoteDataState
 }
 
-class CSVExportButton extends PureComponent<StateProps, {}> {
+class CSVExportButton extends PureComponent<Props, State> {
+  private _controller: AbortController
```

Comment:
I've seen this pattern a bit, where we preface private variables with `_`. Is it necessary? In languages that don't have private attrs, it makes sense. In languages that do, usually the pattern I've seen is that the `_` prefix means "I'm not used anywhere, and am just trying to satisfy the compiler."

Reply:
Fair enough. Will remove.